sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/windowing/AfterAllTest.java

```diff
@@ -149,4 +149,22 @@ public void testContinuation() throws Exception {
         afterAll.getContinuationTrigger());
   }
 
+  @Test
+  public void testToString() {
+    Trigger trigger = AfterAll.of(new NamedTrigger("t1"), new NamedTrigger("t2"));
+    assertEquals("AfterAll.of(t1, t2)", trigger.toString());
+  }
+
+  private static class NamedTrigger extends StubTrigger {
```

Comment:
Make this a class in the test directory here, so that you don't need to recreate NamedTrigger so many times. Alternatively:

```AfterAll.of(
     new StubTrigger() { @Override public String toString() { return "t1"; } },
     new StubTrigger() { @Override public String toString() { return "t2"; } })
```

Or make `StubTrigger` take a name (optionally)?